### PR TITLE
Output last line terminator option for CsvWriter class

### PR DESCRIPTION
**Pull request by blackmo18**
Issue #18 
 - Added Option to CsvWriterContext
 - Added State and State Handler in CsvWriter
     * this due to in CsvWriter open method it is allowed to write in multiple ways and order of:
         1. writeRow
         2. writeRows
- Added Test

### CODE REVIEW

File: src/jvmMain/kotlin/com/github/doyaaaaaken/kotlincsv/client/CsvFileWriter.kt [hunk 1]
```diff
@@ -41,7 +58,14 @@ class CsvFileWriter internal constructor(
      * write rows from Sequence
      */
     override fun writeRows(rows: Sequence<List<Any?>>) {
-        rows.forEach { writeNext(it) }
+        val size = rows.count()
```
Comment (doyaaaaaken) [inline src/jvmMain/kotlin/com/github/doyaaaaaken/kotlincsv/client/CsvFileWriter.kt:61]: `Sequence.count` operation is terminal, so all rows are processed on this point.
Kotlin Doc about `Sequence.count` says
> * Returns the number of elements in this sequence.
> * The operation is _terminal_.

Maybe, `rows.iterator()` is better.

File: src/commonMain/kotlin/com/github/doyaaaaaken/kotlincsv/client/CsvWriterState.kt [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+/**
+ * CSV writer state and handler
+ * @author blacmko18
+ */
+sealed class CsvWriterState(val wroteFirstLine: Boolean, val wroteLastLineTerminator: Boolean)
+
+object DefaultState: CsvWriterState(wroteFirstLine = false, wroteLastLineTerminator =  false)
+object HasNotWroteLastLineTerminator: CsvWriterState(true, wroteLastLineTerminator = false)
+object HasWroteLastLineTerminator: CsvWriterState(true, wroteLastLineTerminator = true)
```
Comment (doyaaaaaken) [inline src/commonMain/kotlin/com/github/doyaaaaaken/kotlincsv/client/CsvWriterState.kt:11]: The word `LastLine` is misleading. 
`LineEnd` would be more appropriate.

File: src/jvmTest/kotlin/com/github/doyaaaaaken/kotlincsv/client/CsvWriterTest.kt [hunk 3]
```diff
@@ -194,6 +194,51 @@ class CsvWriterTest : WordSpec() {
...
+                val row6 = listOf("3", "4")
+                val expected = "a,b\r\nc,d\r\ne,f\r\ng,h\r\n1,2\r\n3,4"
+                csvWriter{
+                    outputLastLineTerminator = false
+                }.open(File(testFileName)) {
+                    writeRow(row1)
+                    writeRows(listOf(row2, row3))
+                    writeRow(row4)
+                    writeRows(listOf(row5, row6))
+
```
[the remark this answers is not in the document]
Comment (blackmo18) [reply src/jvmTest/kotlin/com/github/doyaaaaaken/kotlincsv/client/CsvWriterTest.kt:237]: I think to rewrite all in 1 line defeats the purpose of testing/covering the `CsvWriterState`

[the remark this answers is not in the document]
Comment (doyaaaaaken) [reply src/jvmTest/kotlin/com/github/doyaaaaaken/kotlincsv/client/CsvWriterTest.kt:237]: OK, then not need to rewrite. 👍

File: src/commonMain/kotlin/com/github/doyaaaaaken/kotlincsv/client/CsvWriterState.kt [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+class CsVWriterStateHandler {
```
[the remark this answers is not in the document]
Comment (blackmo18) [reply src/commonMain/kotlin/com/github/doyaaaaaken/kotlincsv/client/CsvWriterState.kt]: fixed at  e99bb94

File: src/commonMain/kotlin/com/github/doyaaaaaken/kotlincsv/client/CsvWriterState.kt [hunk 5]
```diff
@@ -0,0 +1,31 @@
+package com.github.doyaaaaaken.kotlincsv.client
+
+/**
+ * CSV writer state and handler
+ * @author blacmko18
+ */
+sealed class CsvWriterState(val wroteFirstLine: Boolean, val wroteLastLineTerminator: Boolean)
+
+object DefaultState: CsvWriterState(wroteFirstLine = false, wroteLastLineTerminator =  false)
```
[the remark this answers is not in the document]
Comment (blackmo18) [reply src/commonMain/kotlin/com/github/doyaaaaaken/kotlincsv/client/CsvWriterState.kt]: fixed at  e99bb94

File: src/commonMain/kotlin/com/github/doyaaaaaken/kotlincsv/client/CsvWriterState.kt (hunk 4 above)
[the remark this answers is not in the document]
Comment (blackmo18) [reply src/commonMain/kotlin/com/github/doyaaaaaken/kotlincsv/client/CsvWriterState.kt]: fixed at  e99bb94

File: src/jvmMain/kotlin/com/github/doyaaaaaken/kotlincsv/client/CsvFileWriter.kt (hunk 1 above)
Comment (blackmo18) [reply src/jvmMain/kotlin/com/github/doyaaaaaken/kotlincsv/client/CsvFileWriter.kt]: change at 59297f1, tho not sure if this is what you meant. Please give me your thoughts about this

File: src/commonMain/kotlin/com/github/doyaaaaaken/kotlincsv/client/CsvWriterState.kt (hunk 2 above)
Comment (blackmo18) [reply src/commonMain/kotlin/com/github/doyaaaaaken/kotlincsv/client/CsvWriterState.kt]: change at e99bb94

File: src/jvmMain/kotlin/com/github/doyaaaaaken/kotlincsv/client/CsvFileWriter.kt (hunk 1 above)
Comment (doyaaaaaken) [reply src/jvmMain/kotlin/com/github/doyaaaaaken/kotlincsv/client/CsvFileWriter.kt]: Yes, that's correct

Comment (blackmo18) [reply src/jvmMain/kotlin/com/github/doyaaaaaken/kotlincsv/client/CsvFileWriter.kt]: Thanks much appreciated.

**Pull request merged**